<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1">


<!--
Font-awesome icons ie github or twitter
-->
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/all.css" integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf" crossorigin="anonymous">
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/brands.css" integrity="sha384-n9+6/aSqa9lBidZMRCQHTHKJscPq6NW4pCQBiMmHdUCvPN8ZOg2zJJTkC7WIezWv" crossorigin="anonymous">

<!--
Google fonts api stuff
-->
<link href='https://fonts.googleapis.com/css?family=Arial' rel='stylesheet'>
<link href='https://fonts.googleapis.com/css?family=Arial' rel='stylesheet'>



<title> Methylator - DNA methylation pipeline for plants and mammals</title>


<script src="poster_files/kePrint-0.0.1/kePrint.js"></script>
<link href="poster_files/lightable-0.0.1/lightable.css" rel="stylesheet" />




<style>
@page {
size: 841mm 1189mm;
margin: 0;
padding: 0;
}
body {
margin: 0;
font-size: 30px;
width: 841mm;
height: 1189mm;
padding: 0;
text-align: justify;
font-family: Arial;
}
.poster_wrap {
width: 841mm;
height: 1189mm;
padding: 0cm;
}
.title_container {
width: 841mm;
height: calc(1189mm * 0.15);
overflow: hidden;
background-color: #0073b7;
border: 1cm solid #005587;
}
.logo_left {
float: left;
width: 10%;
height: 100%;
background-color: #0073b7;
display: flex;
align-items: center;
justify-content: center;
}
.logo_right {
float: right;
width: 10%;
height: 100%;
background-color: #0073b7;
display: flex;
align-items: center;
justify-content: center;
}
.poster_title {
text-align: center;
position: relative;
float: left;
width: 80%;
height: 100%;
color: #ffffff;
top: 50%;
transform: translateY(-50%);
-webkit-transform: translateY(-50%);
}
#title {
font-family: Arial;
}
/* unvisited link */
a:link {
color: #0073b7;
}
.mybreak {
  break-before: column;
}
/* visited link */
a:visited {
color: #0073b7;
}

/* mouse over link */
a:hover {
color: #0073b7;
}

/* selected link */
a:active {
color: #0073b7;
}
.poster_body {
-webkit-column-count: 3; /* Chrome, Safari, Opera */
-moz-column-count: 3; /* Firefox */
column-count: 3;
-webkit-column-fill: auto;
-moz-column-fill: auto;
column-fill: auto;
-webkit-column-rule-width: 1mm;
-moz-column-rule-width: 1mm;
column-rule-width: 1mm;
-webkit-column-rule-style: dashed;
-moz-column-rule-style: dashed;
column-rule-style: dashed;
-webkit-column-rule-color: #005587;
-moz-column-rule-color: #005587;
column-rule-color: #005587;
column-gap: 1cm;
padding-left: calc(1cm / 2);
padding-right: calc(1cm / 2);
height: 100%;
color: #000000
background-color: #ffffff;
}
.poster_title h1 {
font-size: 95pt;
margin: 0;
border: 0;
font-weight: normal;
}
.poster_body_wrap{
width: calc(841mm + 1cm + 1cm);
height: calc(1189mm * 0.83);
padding-top: calc(1189mm * 0.005);
padding-bottom: calc(1189mm * 0.01);
background-color: #ffffff;
}
.poster_title h3 {
color: #000066;
font-size: 40pt;
margin: 0;
border: 0;
font-weight: normal;
}
.poster_title h3 > sup {
  font-size: 35pt;
  margin-left: 0.02em;
}
.poster_title h5 {
color: #FFFFFF;
font-size: ;
margin: 0;
border: 0;
font-weight: normal;
}
img {
margin-top: 2cm;
margin-bottom: 0;
}
.section {
  padding: 0.2em;
}
.poster_body h1 {
text-align: center;
color: #ffffff;
font-size: 45pt;
border: 1mm solid #005587;
background-color: #005587;
border-radius: 4mm 0mm;
margin-top: 2mm;
margin-bottom: 2mm;
font-weight: normal;
}
.poster_body h2 {
color: #005587;
font-size: 35pt;
padding-left: 4mm;
font-weight: normal;
}
.span {
width: 200%;
}
/* center align leaflet map,
from https://stackoverflow.com/questions/52112119/center-leaflet-in-a-rmarkdown-document */
.html-widget {
margin: auto;
position: sticky;
margin-top: 2cm;
margin-bottom: 2cm;
}
.leaflet.html-widget.html-widget-static-bound.leaflet-container.leaflet-touch.leaflet-fade-anim.leaflet-grab.leaflet-touch-drag.leaflet-touch-zoom {
position: sticky;
width: 100%;
}
pre.sourceCode.r {
background-color: #dddddd40;
border-radius: 4mm;
padding: 4mm;
width: 75%;
/* align-items: center; */
margin: auto;
padding-left: 2cm;
}
code.sourceCode.r{
background-color: transparent;
font-size: 20pt;
border-radius: 2mm;
}
.caption {
font-size: 25pt;
}
.table caption {
font-size: 25pt;
padding-bottom: 3mm;

}
code {
font-size: 1em;
font-family: monospace;
background-color: #00558724;
color: #0073b7;
padding: 1.2mm;
border-radius: 2mm;
}
.poster_title code {
font-size: 1em;
}
table {
font-size: 40px;
margin: auto;
border-top: 3px solid #666;
border-bottom: 3px solid #666;
}
table thead th {
border-bottom: 3px solid #ddd;
}
td {
padding: 8px;
}
th {
padding: 15px;
}
caption {
margin-bottom: 10px;
}
.poster_body p {
margin-right: 4mm;
margin-left: 4mm;
margin-top: 6mm;
margin-bottom: 10mm;
}
.poster_body ol {
margin-right: 4mm;
margin-left: 4mm;
}
#ul {
margin-right: 4mm;
margin-left: 4mm;
}
.references p {
font-size: 20pt;
}
.orcid img {
  width: 1em;
}
</style>
</head>
<body>


<div class="poster_wrap">
<div class="title_container">
<!-- Left Logo  -->
<div class="logo_left">
<img src= 'figures/output-onlinepngtools.png' style="width: 80%">
</div>
<!-- Poster Title -->
<div class= "poster_title">
<h1 id="title"><span class="brsmall"></span> Methylator - DNA methylation pipeline<br>for plants and mammals</h1>
<h3 id="author">Jonas Bucher<sup>1 <i style="font-size:44px" class="fa"></i></sup>, Masaomi Hatakeyama<sup>2,3</sup>, Ueli Grossniklaus<sup>1,4</sup>, Deepak Tanwar<sup>1,4 <i style="font-size:44px" class="fa"></i></sup></h3><br>
<h5 id="affiliation"><sup>1</sup> Plant Development Genetics, Department of Plant and Microbial Biology, University of Zurich<br> <sup>2</sup> Evolutionary and Ecological Genomics, Department of Evolutionary Biology and Environmental Studies<br> <sup>3</sup> Functional Genomics Center Zurich, ETH Zürich and University of Zurich<br> <sup>4</sup> URPP Evolution in Action, Department of Plant and Microbial Biology, University of Zurich<br><br><i style="font-size:24px" class="fa"></i> jonas.bucher<span>@</span>uzh.ch | deepak.tanwar<span>@</span>evolution.uzh.ch</h5>
</div>
<!-- Right Logo  -->
<div class="logo_right">
<img src=figures/QR_800.png style="width: 80%">
</div>
</div>

<div class='poster_body_wrap'>
<div class='poster_body'>
<style>
.mybreak {
   break-before: column;
}
</style>
<style>
.ref-align {
  text-align: left
}
</style>
<style>
img {
  display: block;
  margin-top: -7px; /* Adjust this value as needed */
}

.my-image {
  margin-top: -1px; /* Adjusted value for a specific image */
}
</style>
<style type="text/css">
    .brsmall {
        display: block;
        margin-bottom: .3em;
    }
    .brxsmall {
        display: block;
        margin-bottom: .4em;
    }
    .brsmallneg {
        display: block;
        margin-bottom: -.4em;
    }
</style>
<style type="text/css">
div.logo_left{
  width: 15%;
}
div.poster_title{
  width: 70%;
}
</style>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>DNA Methylation is the addition of a methyl group to a molecule in the DNA. It impacts transcription and therefore plays a major role in several vital processes. In mammals, methylation predominantly occurs in CG sequence contexts, whereas in plants the CHG and CHH contexts are common as well.</p>
<figure>
<center>
<img src="figures/methylation_contexts.png" width="950" style="display:block">
<br>
<figcaption>
<b>Figure 1:</b> Methylation contexts in mammals and plants.
</figcaption>
</center>
</figure>
<p></b></p>
<p>Various tools have been introduced to facilitate the analysis of DNA methylation data. Usually, they focus on a small part of the workflow, which still leaves users with a considerable amount of work to evaluate appropriate tools, transform intermediate output, and finally generate presentable figures. Additionally, many tools are limited in terms of the input data, only providing support for certain species and/or sequencing methods.</p>
</div>
<div id="pipeline-overview" class="section level1">
<h1>Pipeline overview</h1>
<p>Here we introduce Methylator, a user-friendly tool for a full DNA methylation analysis, with a easy-to-use interface, facilitated reproducibility and interactive visualizations of results.
<!-- <div style="line-height:30%; visibility:hidden;"> -->
<!--     <br> -->
<!-- </div> --></p>
<figure>
<center>
<img src="figures/workflow_methylation_final.drawio.png" width="100%" style="display:block">
<br>
<figcaption>
<b>Figure 2:</b> Overview of Methylator.
</figcaption>
</center>
</figure>
<p></b></p>
</div>
<div id="duplicate-removal" class="section level1">
<h1>Duplicate removal</h1>
<p>Although technical duplicate reads can arise from different sources, most deduplication tools focus on PCR duplicates. Clumpify can deal with the different types of duplication in the sequencing data, such as optical duplicates. As the abundance of the types of duplicates in the data depends on the sequencing technology used, Methylator adapts the duplication removal accordingly to the user input.</p>
<div id="types-of-duplicates-in-sequencing-data" class="section level2">
<h2>Types of duplicates in sequencing data</h2>
<figure>
<center>
<img src="figures/duplicates_final.drawio.png" width="100%" style="display:block">
<br>
<figcaption>
<b>Figure 3:</b> Types of duplicates that can be present after sequencing. Adapted from <a href="biostars.org/p/229842/">biostars.org/p/229842/</a>
</figcaption>
</center>
</figure>
<p></b></p>
</div>
</div>
<div id="alignment" class="section level1 mybreak">
<h1>Alignment</h1>
<p>Bisulfite treatment of DNA results in a decreased sequence complexity, which deteriorates mapping efficiency. Overall, this causes loss of a big proportion of the sequencing data for analysis.</p>
<div id="dirty-harry-method" class="section level2">
<h2>Dirty harry method</h2>
<p>The Dirty harry method offers an improvement to the mapping rate by remapping the unaligned reads locally.
Through that, this method increases the mapping efficiency and retains a considerable amount of cytosine sites, which would otherwise be lost.</p>
<figure>
<center>
<img src="figures/dirty_harry_new.drawio.png" width="950" style="display:block">
<br>
<figcaption>
<b>Figure 4:</b> Dirty harry alignment method. Adapted from <span class="citation">Wu et al. (<a href="#ref-Wu2019" role="doc-biblioref">2019</a>)</span>
</figcaption>
</center>
</figure>
<p></b></p>
</div>
</div>
<div id="visualization" class="section level1">
<h1>Visualization</h1>
<p>For each type of analysis, several outputs are generated and visualized in a single interactive shiny app.
Publication-ready figures are created using colourblind-friendly palettes. Each plot can be customized and downloaded individually by the user.</p>
<!-- # ```{r, shiny_1, fig.show="hold", out.width="20%", fig.cap='Colour option example for a PCA plot in...'} -->
<!-- # knitr::include_graphics("~/git/poster_urpp_conference/figures/colourOptions.png") -->
<!-- # knitr::include_graphics("~/git/poster_urpp_conference/figures/dotPlot.png") -->
<!-- # ``` -->
<figure>
<center>
<img src="figures/enrichmentTable_full.png" width="95%" style="display:block">
<br>
<figcaption>
<b>Figure 5:</b> Enrichtment table.
</figcaption>
</center>
</figure>
<p></b></p>
<figure>
<center>
<img src="figures/dotPlot_full2.png" width="100%" style="display:block">
<br>
<figcaption>
<b>Figure 6:</b> Dot plot of enrichment of genes in biological processes.
</figcaption>
</center>
</figure>
<p></b></p>
<figure>
<center>
<img src="figures/volcanoPlot.png" width="950" style="display:block">
<br>
<figcaption>
<b>Figure 7:</b> Volcano Plot of genomic region enrichment.
</figcaption>
</center>
</figure>
<p></b></p>
<div id="visual-customization" class="section level2">
<h2><span class="brsmallneg"></span> Visual customization</h2>
<figure>
<center>
<img src="figures/col_merged.png" width="100%" style="display:block">
<br>
<figcaption>
<b>Figure 8:</b> Colour customization of figures.
</figcaption>
</center>
</figure>
<p></b></p>
<!-- Dot plot of enrichment of genes in biological processes. -->
<!-- Example of colour options for plotting, with pre-defined colourblind-friendly palettes and manual input. -->
<!-- <img align="left" src="figures/dotPlot.png" width="35%" hspace="20"/> -->
<!-- <img align="left" src="figures/colourOptions.png" width="15%" hspace="20"/> -->
<!-- <figure> -->
<!-- <center> -->
<!--   <img src="figures/enrichmentTable.png" width="650" style="display:block"> -->
<!--   <br> -->
<!--   <figcaption><b>Figure 6:</b> Enrichment Table.</figcaption> -->
<!-- </center> -->
<!-- </figure> -->
<!-- </b> -->
<!-- <img width="20%" style="float: right;" src="figures/colourOptions.png"/> -->
</div>
</div>
<div id="comparison-with-other-tools" class="section level1">
<h1>Comparison with other tools</h1>
<style>
.table>tbody>tr>td{
  padding: 1px; border: hidden;
}
</style>
<table class="table" style="font-size: 23px; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;font-weight: bold;border-bottom: 0px solid">
</th>
<th style="text-align:center;font-weight: bold;border-bottom: 0px solid">
Methylator
</th>
<th style="text-align:center;font-weight: bold;border-bottom: 0px solid">
methylseq
</th>
<th style="text-align:center;font-weight: bold;border-bottom: 0px solid">
ARPEGGIO
</th>
<th style="text-align:center;font-weight: bold;border-bottom: 0px solid">
MethylStar
</th>
<th style="text-align:center;font-weight: bold;border-bottom: 0px solid">
MethylC-analyzer
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;background-color: #CCE5FF !important;font-weight: bold;">
Platform independent
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✓
</h3>
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✓
</h3>
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✓
</h3>
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✕
</h3>
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Self-contained
</td>
<td style="text-align:center;">
<h3>
✓
</h3>
</td>
<td style="text-align:center;">
<h3>
✓
</h3>
</td>
<td style="text-align:center;">
<h3>
✓
</h3>
</td>
<td style="text-align:center;">
<h3>
✓
</h3>
</td>
<td style="text-align:center;">
<h3>
✕
</h3>
</td>
</tr>
<tr>
<td style="text-align:left;background-color: #CCE5FF !important;font-weight: bold;">
Interface
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
GUI (SUSHI, Galaxy, Shiny), CLI
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
CLI
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
CLI
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
CLI
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
CLI, GUI
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Input Data
</td>
<td style="text-align:center;">
WGBS, RRBS, PBAT, TAPS, ABBS
</td>
<td style="text-align:center;">
WGBS, RRBS
</td>
<td style="text-align:center;">
WGBS
</td>
<td style="text-align:center;">
WGBS, PBAT, single-cell
</td>
<td style="text-align:center;">
WGBS, RRBS
</td>
</tr>
<tr>
<td style="text-align:left;background-color: #CCE5FF !important;font-weight: bold;">
Supported genomes
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
Mammals, Plants (incl. Polyploids)
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
General
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
Polyploids
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
Mammals, Plants
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
Mammals, Plants
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Quality control
</td>
<td style="text-align:center;">
<h3>
✓
</h3>
</td>
<td style="text-align:center;">
<h3>
✓
</h3>
</td>
<td style="text-align:center;">
<h3>
✓
</h3>
</td>
<td style="text-align:center;">
<h3>
✓
</h3>
</td>
<td style="text-align:center;">
<h3>
✕
</h3>
</td>
</tr>
<tr>
<td style="text-align:left;background-color: #CCE5FF !important;font-weight: bold;">
Alignment
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
Bismark (incl. dirty harry),
Arioc (GPU-based), EAGLE-RC
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
Bismark, bwa-meth
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
Bismark, EAGLE-RC
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
Bismark
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✕
</h3>
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Deduplication
</td>
<td style="text-align:center;">
Clumpify
</td>
<td style="text-align:center;">
Bismark, Picard
</td>
<td style="text-align:center;">
Bismark
</td>
<td style="text-align:center;">
Bismark
</td>
<td style="text-align:center;">
<h3>
✕
</h3>
</td>
</tr>
<tr>
<td style="text-align:left;background-color: #CCE5FF !important;font-weight: bold;">
Methylation context
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
All
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
All
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
All
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
All
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
All
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Exploratory data analysis
</td>
<td style="text-align:center;">
PCA, heatmaps, methylation summaries
</td>
<td style="text-align:center;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;">
PCA, heatmaps,
methylation summaries
</td>
</tr>
<tr>
<td style="text-align:left;background-color: #CCE5FF !important;font-weight: bold;">
Differential methylation analysis
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
DMRs, DMLs
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✓
</h3>
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
DMRs and DMGs
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Copy number variation analysis
</td>
<td style="text-align:center;">
CNVkit
</td>
<td style="text-align:center;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;">
<h3>
✕
</h3>
</td>
</tr>
<tr>
<td style="text-align:left;background-color: #CCE5FF !important;font-weight: bold;">
Functional analysis
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
GO, Kegg, Reactome, user-defined
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✕
</h3>
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Motif analysis
</td>
<td style="text-align:center;">
Homer
</td>
<td style="text-align:center;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;">
<h3>
✕
</h3>
</td>
</tr>
<tr>
<td style="text-align:left;background-color: #CCE5FF !important;font-weight: bold;">
Visualization
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✓
</h3>
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✕
</h3>
</td>
<td style="text-align:center;background-color: #CCE5FF !important;">
<h3>
✓
</h3>
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Year published
</td>
<td style="text-align:center;">
In development
</td>
<td style="text-align:center;">
2020
</td>
<td style="text-align:center;">
2021
</td>
<td style="text-align:center;">
2020
</td>
<td style="text-align:center;">
2023
</td>
</tr>
</tbody>
</table>
<p>      <b>Table 1:</b> Comparison with features of popular tools.<br></p>
<!-- <b>Table 1:</b> Comparison with features of popular tools.<br><br> -->
<p>GUI = graphical user interface, CLI = command line interface<br>
DMR/DML/DMG = Differentially Methylated Regions/Loci/Genes<br>
GO = Gene Ontology</p>
<!-- <div style="line-height:30%; visibility:hidden;"> -->
<!--     <br> -->
<!-- </div> -->
</div>
<div id="conclusion" class="section level1">
<h1>Conclusion</h1>
<!-- * User-friendly methylation analysis pipeline -->
<!-- * Support for plants & mammals -->
<!-- * Interactive visualizations -->
<!-- * Reproducibility through docker -->
<div id="user-friendly-reproducible-and-sustainable-methylation-data-analysis-pipeline-support-for-plants-mammals-can-analyze-metylation-data-from-various-library-preparation-methods-for-bulk-sequencing-interactive-visualizations-of-results" class="section level2">
<h2>User-friendly, reproducible and sustainable methylation data analysis pipeline<br><br>Support for plants &amp; mammals<br><br>Can analyze metylation data from various library preparation methods for bulk sequencing <br><br>Interactive visualizations of results</h2>
</div>
</div>
<div id="references" class="section level1 ref-align">
<h1>References</h1>
<p><span class="brsmall"></span></p>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Wu2019" class="csl-entry">
Wu, Peng, Yan Gao, Weilong Guo, and Ping Zhu. 2019. <span>“Using Local Alignment to Enhance Single-Cell Bisulfite Sequencing Data Efficiency.”</span> <em>Bioinformatics</em> 35 (September): 3273–78. <a href="https://doi.org/10.1093/bioinformatics/btz125">https://doi.org/10.1093/bioinformatics/btz125</a>.
</div>
</div>
</div>
</div>
</div>

</div>



</body>
</html>
